src/tiledbsoma_ml/_utils.py

```diff
@@ -6,13 +6,22 @@
 import itertools
 import sys
 from itertools import islice
-from typing import Any, Iterable, Iterator, Tuple, TypeVar
+from typing import Iterable, Iterator, Tuple, TypeVar
 
 import numpy as np
 import numpy.typing as npt
 
+try:
```

Comment:
_EagerIter is such a tiny bit of code, I suggest you just vendor it into this repo.  It is not a public interface of either `somacore` or `tiledbsoma`, which makes it even more compelling to cut&paste

Reply:
I stand by my recommendation - it is private, small and simple, so I would vendor it. Otherwise, you are effectively making it a public API for tiledbsoma, with all that that entails (i.e, documentation, etc).

Reply:
OK, that makes sense, I copied in `{test,}_eager_iter.py`, and updated the import in `dataset.py`.